internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -2237,11 +2293,51 @@ func getStorageRequests(pvcSpec *corev1.PersistentVolumeClaimSpec) map[corev1.Re
 	return requests
 }
 
-func (r *AgentServiceConfigReconciler) getImageService(ctx context.Context, log logrus.FieldLogger) string {
-	imageServiceURL, err := r.urlForRoute(ctx, imageServiceName)
+func (r *AgentServiceConfigReconciler) getImageService(ctx context.Context, log logrus.FieldLogger, instance *aiv1beta1.AgentServiceConfig) string {
+	imageServiceURL, err := r.urlForRoute(ctx, imageServiceName, instance)
 	if err != nil {
 		log.WithError(err).Warnf("Failed to get URL for route %s", imageServiceName)
 		return ""
 	}
 	return imageServiceURL
 }
+
+func (r *AgentServiceConfigReconciler) deployAgentInstallCRDs(secret *corev1.Secret, namespace string, log logrus.FieldLogger) error {
+	// Get in-cluster client
+	kubeClient, err := r.K8sApiExtensionsClientFactory.CreateFromInClusterConfig()
+	if err != nil {
+		return err
+	}
+
+	// Fetch all agent-install CRDs in cluster
+	crdClient := kubeClient.ApiextensionsV1().CustomResourceDefinitions()
+	crds, err := crdClient.List(context.TODO(), metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("operators.coreos.com/assisted-service-operator.%s=", namespace),
+	})
+	if err != nil || len(crds.Items) == 0 {
+		log.Error(err)
+		return pkgerror.New("agent-install CRDs are not available")
+	}
+
+	// Get external cluster client (using kubeconfig specified in ASC)
+	kubeClient, err = r.K8sApiExtensionsClientFactory.CreateFromSecret(secret)
+	if err != nil {
+		return err
+	}
+
+	// Create the CRDs on the external cluster
```

Comment:
This whole function is looking very imperative, even from the name "deploy...". When reconciling, we want a declarative pattern that:
1. figures out the desired state
1. inspects the real world to identify the current state
1. changes the real world to bring it closer to the desired state

Per comments elsewhere, we should figure out a coordinated strategy with ACM and hive for how to install and manage the CRDs in the hypershift control planes. If we do it through code like this, we'll probably want to setup a long-lived dynamic client for each control plane that can be reused.